### PR TITLE
fix: Resolved Typescript Error in PR #9598 created by @vijeet-shah.

### LINKED ISSUE #9539: [CAL-1955] Possible conditional hook usage issue
Error while testing the user onboarding locally. Can be replicated by creating a new user.

We should make that usage unconditional.
cc @sean-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon-brydon

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1955](https://linear.app/calcom/issue/CAL-1955/possible-conditional-hook-usage-issue)</sub>

### PR DESCRIPTION
**Pull request by pateldivyesh1323**
## What does this PR do?

Error while testing the user onboarding locally. Can be replicated by creating a new user.
Removed the conditional assignment, the queryAvailability hook will always be used, ensuring that the availability data is consistently fetched and assigned to the availabilityForm.

Fixes #9539 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
The issue was already solved by @vijeet-shah. Resolved Typescript Error in PR#9598.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by pateldivyesh1323** [comment]
@keithwillcode Can You Please review this. I have solved the type error created in pr #9598 which solved issue #9539

File: apps/web/components/getting-started/steps-views/SetupAvailability.tsx [hunk 1]
```diff
@@ -22,15 +22,13 @@ const SetupAvailability = (props: ISetupAvailabilityProps) => {
...
-    queryAvailability = trpc.viewer.availability.schedule.get.useQuery(
-      { scheduleId: defaultScheduleId },
-      {
-        enabled: router.isReady,
-      }
-    );
-  }
+
+  const queryAvailability = trpc.viewer.availability.schedule.get.useQuery(
+    { scheduleId: defaultScheduleId as number | undefined },
```
Comment (Udit-takkar) [inline apps/web/components/getting-started/steps-views/SetupAvailability.tsx:27]: defaultScheduleId can still be null which can cause this error.

Screenshot 2023-08-22 at 1 50 13 AM

File: apps/web/components/getting-started/steps-views/SetupAvailability.tsx [hunk 2]
```diff
@@ -22,15 +22,13 @@ const SetupAvailability = (props: ISetupAvailabilityProps) => {
...
+    {
+      enabled: router.isReady,
+    }
+  );
```
Comment (Udit-takkar) [inline apps/web/components/getting-started/steps-views/SetupAvailability.tsx:31]:
```suggestion
  const scheduleId = defaultScheduleId === null ? undefined : defaultScheduleId;
  const queryAvailability = trpc.viewer.availability.schedule.get.useQuery(
    { scheduleId },
    {
      enabled: router.isReady && !!scheduleId ,
    }
  );
```

**Comment by Udit-takkar** [comment]
@pateldivyesh1323 format using prettier and resolve the conflicts

**Review by Udit-takkar — APPROVED** [review]

**Comment by pateldivyesh1323** [comment]
@Udit-takkar I think you need to give approval

**Comment by pateldivyesh1323** [comment]
Thankyou everyone for help.This was my first PR

**Pull request merged**